src/riscv/lib/src/jit.rs

```diff
@@ -348,75 +351,82 @@ mod tests {
             }
         }
 
+        fn check_compilable(&self) {
+            // Ensure the set of instructions can be compiled in JIT.
+            let mut test_jit = JIT::<M4K>::new().unwrap();
+            test_jit
+                .compile(&self.instructions, self.initial_pc.unwrap_or_default())
+                .expect("JIT compilation should succeed.");
+        }
+
         /// Run a test scenario over both the Interpreted & JIT modes of compilation,
         /// to ensure they behave identically.
-        fn run(&self, jit: &mut JIT<M4K>, interpreted_bb: &mut InterpretedBlockBuilder) {
-            // Create the states for the interpreted and jitted runs.
-            let mut interpreted = MachineCoreState::<M4K, _>::new();
-            interpreted.main_memory.set_all_readable_writeable();
+        fn run(&self) {
+            // ensure the set of instructions can be compiled in JIT.
+            self.check_compilable();
 
-            let mut jitted = MachineCoreState::<M4K, _>::new();
-            jitted.main_memory.set_all_readable_writeable();
+            // Create the states for the interpreted and jitted runs.
+            let mut interpreted_state: MachineState<
+                M4K,
+                TestCacheConfig,
+                Interpreted<M4K, Owned>,
+                Owned,
+            > = MachineState::new(InterpretedBlockBuilder);
+            interpreted_state
+                .core
+                .main_memory
+                .set_all_readable_writeable();
+
+            let mut jitted_state: MachineState<
+                M4K,
+                TestCacheConfig,
+                Jitted<InlineCompiler<M4K>, M4K>,
+                Owned,
+            > = MachineState::new(InlineCompiler::default());
+            jitted_state.core.main_memory.set_all_readable_writeable();
```

Comment:
Bit of a nit, to clear boilerplate. Works together with adding this type near the other type defs in the module
```
    /// Machine state for test scenarios, can still configure the [`Block`] type.
    type TestMachineState<B> = MachineState<M4K, TestCacheConfig, B, Owned>;
```


```suggestion
            // Create the states for the interpreted and jitted runs.
            let mut interpreted_state: TestMachineState<Interpreted<_, _>> =
                MachineState::new(InterpretedBlockBuilder);
            interpreted_state
                .core
                .main_memory
                .set_all_readable_writeable();

            let mut jitted_state: TestMachineState<Jitted<InlineCompiler<_>, _>> =
                MachineState::new(InlineCompiler::default());
            jitted_state.core.main_memory.set_all_readable_writeable();
```

Reply:
thanks! done [here](https://github.com/tezos/riscv-pvm/compare/26d7725d8a14caa5bf0e5b12bc6dac8f8a4b1eec..2db4038956e35b131994323ad5bc7281ef24317d)